x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -380,11 +380,13 @@ export interface ActionListApiResponse {
   userIds: string[] | undefined; // users that requested the actions
   commands: string[] | undefined; // type of actions
   /**
-   * The `outputs` is not currently part of the list response due to possibly large amounts of
+   * Introduced in 8.8, outputs are visible for specific actions like `execute`.
+   * The `outputs` are part the list response for given set of actions.
+   * `outputs` for all actions are restricted due to possibly large amounts of
    * data, especially for cases (in the future) where we might support actions being sent to
    * multiple agents
    */
-  data: Array<Omit<ActionDetails, 'outputs'>>;
+  data: ActionDetails[];
```

Comment:
This is not correct. The data can be mixed and it should be reflected here. So the type should be:

```typescript
data: Array<ActionDetails | Omit<ActionDetails, 'outputs'>>
```

Maybe create a new type for the one that does not have `outputs`: 

```typescript
/** Action Details normally returned by the Action List API */
type ActionDetailsNoOputputs = Omit<ActionDetails, 'outputs'>;
```

And then use this in the `data` type above. With this new type, you can then also build a quick TS type guard that can be used to narrow the type (but might not be needed). something like:

```typescript
export const isActionDetailsWithOutputs = (action: ActionDetails | ActionDetailsNoOutputs): action is ActionDetails => {
    return ('outputs' in action);
}
```

Reply:
3599fc0956cbd07acf1f938ebd05f7677e15263a

Reply:
minor adjustment to types e43852c17fc1ebd723294ed8a01fe70e9f9c0cbd to fix https://buildkite.com/elastic/kibana-pull-request/builds/107811#01864ef9-2a38-4c36-bfee-3d041c1e1142